docs/source/configuration.rst

```diff
@@ -350,6 +350,79 @@ policy.
 
         *"It's some arcane magic from AT LEAST a decade ago."*
 
+Loop prevention
+---------------
+
+In order to prevent the bot from entering a loop (for example when there is
+another bot in the same channel, or if a user spams a command), it'll try to
+see if the next message to send is repeating too often in the last ten messages
+in the last few minutes. If that happens, the bot will send ``...`` a few times
+before remaining silent::
+
+    <bot> I repeat myself!
+    <bot> I repeat myself!
+    <bot> I repeat myself!
+    <bot> I repeat myself!
+    <bot> I repeat myself!
+    # wanted to say: "I repeat myself"
+    <bot> ...
+    # wanted to say: "I repeat myself"
+    <bot> ...
+    # wanted to say: "I repeat myself"
+    <bot> ...
+    # silence, wanted to say: "..." instead of "I repeat myself"
+
+This doesn't affect non-repeating messages, and if enough time has passed
+between now and the last message sent, the loop prevention won't be triggered.
+
+This behavior can be configured with:
+
+* :attr:`~CoreSection.antiloop_threshold`: the number of repeating messages
+  before triggering the loop prevention.
+* :attr:`~CoreSection.antiloop_silent_after`: how many times the bot will send
+  the repeat text until it remains silent.
+* :attr:`~CoreSection.antiloop_window`: how much time (in seconds) since the
+  last message must pass before ignoring the loop prevention.
+* :attr:`~CoreSection.antiloop_repeat_text`: the text used to replace repeating
+  messages (default to ``...``).
+
+For example this configuration::
+
+    [core]
+    antiloop_threshold = 2
+    antiloop_silent_after = 1
+    antiloop_window = 60
+    antiloop_repeat_text = Ditto.
+
+will activate the loop prevention feature if there is at least one message in
+the last 60s, from the last 10 messages, 2 are already the same. In that case
+the bot will send ``...``, but only *once*. After that, the bot will remain
+silent. This doesn't affect other messages, i.e. messages that don't repeat::
```

Comment:
I thought about this one for quite a while, ngl.

```suggestion
will activate the loop prevention feature if there are at least 2 messages
in the last 60 seconds, **and** exactly 2 of those messages are the same.
After sending ``...`` *once* (a third message), the bot will remain silent.

This doesn't affect other messages, i.e. messages that don't repeat::
```

Here again, the 10-message maximum history doesn't matter and can only confuse the reader with too much detail. The `.. note::` below about the maximum value of `antiloop_threshold` covers it enough, IMO.

Can a highly active Sopel instance exhibit behavior that contradicts this simplified explanation of how loop prevention works? Of course it can. Running more than ten different commands in one minute, all with different output, and then repeating one that has been pushed off the top of the history list will make it appear that loop prevention "doesn't work", but that's not a typical case.

(I think it usually goes without saying, but I'll say it anyway: If my reasoning doesn't make sense, _please do_ tell me instead of just accepting the suggestion anyway, so I can suggest another way of clarifying what's being said here without losing as many details.)

Reply:
No you are absolutely right: it makes sense to *me* because I'm knee deep into that ~~garbage~~ code, however I realize it's not going to help the reader. Good catch!

(thanks for taking the time to check on that) (your reasoning makes a lot of sense to me)